lib/screens/app/wallet/scan.dart

```diff
@@ -104,56 +50,94 @@ class _ScanState extends State<Scan> {
               account: action.account,
               name: action.name,
               data: data,
-            ), true);
+            ),
+            true);
       } else {
         print("unable to read QR, continuing");
-        scan();
+        setState(() {
+          _handledQrCode = false;
+          this.step = Steps.scan;
+        });
       }
     } catch (e) {
       print("scan error: " + e);
       setState(() {
-        this.error = 'Invalid QR code: $e';
+        this.error = 'Invalid QR code';
         this.step = Steps.error;
       });
     }
   }
 
+  void _showToast(BuildContext context, String message) {
+    _scaffoldKey.currentState.showSnackBar(SnackBar(
+      content: Text(message),
+      duration: Duration(seconds: 3),
+    ));
+  }
+
+  Future<void> _onQRViewCreated(QRViewController controller) async {
+    this.controller = controller;
+
+    controller.scannedDataStream.listen(
+      (String scanResult) async {
+        if (_handledQrCode) {
+          return;
+        }
+
+        _handledQrCode = true;
+
+        if (scanResult == null) {
+          Navigator.of(context).pop();
+        } else {
+          setState(() {
+            this.step = Steps.processing;
+            this.qrcode = scanResult;
+          });
+
+          if (widget.shouldSendResultsBack) {
+            Navigator.pop(context, scanResult);
+          } else {
+            // QUESTION: Why do we not check for other URL types like deep link for invite here?
```

Comment:
we dont need invite when already inside of account...

Reply:
True, this was a cope paste. Ill remove